tests/system/large/test_session.py

```diff
@@ -46,7 +50,72 @@
         # ),
     ],
 )
-def test_read_gbq_for_large_tables(session: Session, query_or_table, index_col):
+def test_read_gbq_for_large_tables(
+    session: bigframes.Session, query_or_table, index_col
+):
     """Verify read_gbq() is able to read large tables."""
     df = session.read_gbq(query_or_table, index_col=index_col)
     assert len(df.columns) != 0
+
+
+def test_close(session):
+    # we will create two tables and confirm that they are deleted
+    # when the session is closed
+
+    bqclient = session.bqclient
+
+    expiration = (
+        datetime.datetime.now(datetime.timezone.utc)
+        + bigframes.constants.DEFAULT_EXPIRATION
+    )
+    full_id_1 = bigframes.session._io.bigquery.create_temp_table(session, expiration)
+    full_id_2 = bigframes.session._io.bigquery.create_temp_table(session, expiration)
+
+    with pytest.raises(google.cloud.exceptions.Conflict):
+        bqclient.create_table(full_id_1)
+    with pytest.raises(google.cloud.exceptions.Conflict):
+        bqclient.create_table(full_id_2)
+
+    session.close()
+
+    with pytest.raises(google.cloud.exceptions.NotFound):
+        bqclient.delete_table(full_id_1)
+    with pytest.raises(google.cloud.exceptions.NotFound):
+        bqclient.delete_table(full_id_2)
+
+
+def test_clean_up_by_session_id():
+    original_location = bigframes.options.bigquery.location
+    bpd.close_session()  # close the session to change regions
+    bigframes.options.bigquery.location = "europe-west10"
+    # we do this test in a different region in order to avoid
+    # overly large amounts of temp tables slowing the test down
+
+    session = bpd.get_global_session()
+    session_id = session.session_id
```

Comment:
Using the global session for this is somewhat scary since other tests running in parallel might try to do the same thing.

Can we test what we need to test without using the global session here? If not, maybe we can create a test fixture for the global session that uses a lock so that tests that use the global session don't run in parallel?

Reply:
How about just building a session object and passing in the location in the context parameter? I've made that change. Let me know if you're aware of any problem with it. Thanks!

Reply:
https://github.com/googleapis/python-bigquery-dataframes/pull/652 actually is my approach to addressing this problem. `option_context` can create a thread-local session.